src/scripts/bc-api-frame.js

```diff
@@ -89,35 +89,29 @@ bc.ApiFrame.prototype._receiveApiMessage = function(event) {
 					this._callRestObj(i);
 					i = this.frameLoadQueue.shift();
 				}
-			} else if(this.frameLoadQueue.length > 0 && method === 'reconnected') {
-				bc.util.log('** We are RECONNECTED', false, rest);
-				var i = this.frameLoadQueue.shift();
-				this.isConnectionError = false;
-				var scope = this;
-				while(i) {
-					bc.util.log('** Attempting CallRestObj again', false, i);
-					(function(i) {
-						setTimeout(function() {
-							scope._callRestObj(i);
-						}, 250);
-					})(i);
-					i = this.frameLoadQueue.shift();
-				}
-				this.messageListener(message.method, message.params, message.id);
-			} else if(method === 'reconnecting') {
-				bc.util.log('** RECONNECTING');
+				
+				var context = this;
+				setInterval(function() {
+					var current = new Date().getTime();
+					for (var i in context.framePendingResults) {
+						var pending = context.framePendingResults[i];
+						var time = pending.time;
+						var wait = Math.min(context.retryTimeout * pending.tries, 60000);
+						if (time - current > wait) {
+							context._callRestObj(pending.rest, pending.tries);
+						}
+					}
+				}, this.retryTimeout);
 			} else {
 				this.messageListener(message.method, message.params, message.id);
 			}
 		} else {
 			var rest = this.framePendingResults[message.id];
 			if(rest) {
 				if(!message.error) {
-					this.framePendingResults[message.id] = null;
+					delete this.framePendingResults[message.id];
 					rest.callback.finished(message);
```

Comment:
Can you change `rest.callback.finished(message)` to `rest.rest.callback.finished(message)`